### PR TITLE
Fixed the comparison in the content-type header to be case insensitive

### PR DESCRIPTION
**Pull request by Pacane**

### CODE REVIEW

**Comment by kevmoo** [comment]
LGTM, but @nex3 should verify. Please rebase your commits into one commit.

And thanks1

**Comment by kevmoo** [comment]
Also, update the commit message to include `Closes URL-to-issue` – Github does magic to link these together

**Comment by Pacane** [comment]
@kevmoo There you go!

File: lib/src/message.dart [hunk 1]
```diff
@@ -138,13 +138,16 @@ abstract class Message {
 Map<String, String> _adjustHeaders(
     Map<String, String> headers, Encoding encoding) {
   if (headers == null) headers = const {};
-  if (encoding == null) return headers;
-  if (headers['content-type'] == null) {
-    return addHeader(headers, 'content-type',
+
+  var caseInsensitiveHeaders = new CaseInsensitiveMap.from(headers);
```
Comment (nex3) [inline lib/src/message.dart]: Don't copy the headers map if `encoding` is `null`.

Also, if you just re-assign to `headers`, you can avoid wrapping the const map in the case that `headers` was null.

File: test/message_change_test.dart [hunk 2]
```diff
@@ -59,14 +59,14 @@ void _testChange(Message factory(
     var request = factory(headers: {'header1': 'header value 1'});
     var copy = request.change(headers: {});
 
-    expect(copy.headers, same(request.headers));
```
Comment (nex3) [inline test/message_change_test.dart]: These `same()` checks are important. They test that the headers map isn't copied more than necessary.

File: test/response_test.dart [hunk 3]
```diff
@@ -140,4 +140,41 @@ void main() {
       expect(response.read, throwsStateError);
     });
   });
+
+  group("content type should be preserved when setting encoding", () {
```
Comment (nex3) [inline test/response_test.dart]: These should by in `message_test.dart`.

**Comment by nex3** [comment]
Looking pretty good! This is definitely the right direction, there are just a few changes to make.

**Comment by Pacane** [comment]
There you go @nex3 Thanks for the review.

**Comment by nex3** [comment]
Thanks for the fix!

**Pull request merged**